### PR TITLE
Added support for multiple tool calls, langchain tools

### PR DESCRIPTION
**Pull request by ronfromhp**

### CODE REVIEW

File: openai-assistant/app.py [hunk 1]
```diff
@@ -161,20 +191,20 @@ async def run_conversation(message_from_ui: cl.Message):
...
-                                thread_id=thread.id,
-                                run_id=run.id,
-                                tool_outputs=[
-                                    {
-                                        "tool_call_id": tool_call.id,
-                                        "output": output,
-                                    },
-                                ],
+                            tool_output = tool_map[function_name].invoke(json.loads(tool_call.function.arguments)["__arg1"])
+                            print("ARGUEMENTSSSSS", tool_call.function.arguments)
```
Comment (willydouhard) [inline openai-assistant/app.py:195]: remove print

File: openai-assistant/create_assistant.py [hunk 2]
```diff
@@ -1,31 +1,59 @@
...
 
 Given a formula below $$ s = ut + \frac{1}{2}at^{2} $$ Calculate the value of $s$ when $u = 10\frac{m}{s}$ and $a = 2\frac{m}{s^{2}}$ at $t = 1s$
 """
-    assistant = await client.beta.assistants.create(
-        name="Math Tutor",
-        instructions=instructions,
-        tools=[{"type": "code_interpreter"}],
-        model="gpt-4-1106-preview",
+   
+    assistant = OpenAIAssistantRunnable.create_assistant(
```
Comment (willydouhard) [inline openai-assistant/create_assistant.py:41]: I'd like to keep this example openai only, we might another one for langchain

File: openai-assistant/app.py [hunk 3]
```diff
@@ -107,7 +116,16 @@ async def run_conversation(message_from_ui: cl.Message):
                 await process_thread_message(message_references, thread_message)
 
             if step_details.type == "tool_calls":
+                print("TOOL CALLS", step_details)
                 for tool_call in step_details.tool_calls:
+                    # print("NOW DICT", tool_call)
```
Comment (willydouhard) [inline openai-assistant/app.py:121]: remove prints

**Review by willydouhard — COMMENTED** [review]
Really cool! Few comments to address

File: openai-assistant/create_assistant.py [hunk 4]
```diff
@@ -3,57 +3,113 @@
...
-        model = "gpt-4-1106-preview",
+
+    assistant = await client.beta.assistants.create(
+        name="Math Tutor",
+        instructions=instructions,
+        tools=tools,
+        model="gpt-4-1106-preview",
     )
-    print("langchain assistant created", assistant.assistant_id)
+    print("langchain assistant created", assistant.id)
```
Comment (willydouhard) [inline openai-assistant/create_assistant.py:97]: remove that print

Comment (ronfromhp) [reply openai-assistant/create_assistant.py]: that print was there in the original cookbook but ok, ive removed it now

**Pull request merged**